DMTN-148.tex

```diff
@@ -50,9 +50,255 @@
 % ADD CONTENT HERE
 % You can also use the \input command to include several content files.
 
+\section{Intro}
+
+The goal of this tech note is to provide a plan for the handling of
+calibration products in the Gen-3 Butler environment.  There are a
+large number of these products currently in use, but many of them do
+not have a clearly defined process to define, ingest, test, and
+deprecate them.  The lack of documentation on how these processes work
+has led to many implementations of similar paths, further complicating
+the issue.
+
+
+\section{The End State}
+
+The final state that we wish to get to has the following properties:
+
+\begin{itemize}
+\item All calibration products are either lsst.afw.image.Exposure
+  image data or an explicit calibration class in the \verb|ip_isr|
+  package.\footnote{This will require class reorganization to migrate
+    the Defects and Curve calibrations definitions.}
+\item All calibration products should be constructed from a DM Task in
+  the \verb|cp_pipe| package, and have the input data and
+  configuration parameters persisted with the calibration itself.
+  This guarantees that any product can be reconstructed in the future.
+  Exceptions will exist for external or lab created calibrations,
+  although the majority of used calibrations will likely be produced
+  directly.
+\item These provenance, config, and final product files may be saved
```

Comment:
It is an oft-stated “requirement” (note the scare quotes) that the data in git repositories must be “human readable” or even “human editable”. That's not mentioned here. Is that deliberate?

Reply:
Also, just to be clear — this is requiring that all products, even bulk image data, can be saved to git repositories. Right?

Reply:
No. We aren't advocating writing the bulk image data to git (I mean with LFS you could do that but I haven't heard anyone asking for that).

Reply:
I don't expect that we will push image data to git (but have no objection if we want to).  I tried to keep this section at least to have "require"/"will"/"must" => requirement; "may"/"should" => allowed and supported.
My idea in making things human readable is that also makes them obviously diffable, so we can keep track of value changes and to support the certification process (which I know is also poorly defined).  Evaluating a flat or bias is simple to do just by looking at the image/the results of processing with that product.  Evaluating crosstalk/linearity/brighter fatter isn't always obvious from the results of ip_isr, so my hope is that being able to look at the (reasonably small) set of numbers will work.